--- a/waydroid/src/waydroid/tools/services/user_manager.py	2021-09-18 18:50:25.320230647 +0200
+++ b/waydroid/tools/services/user_manager.py	2021-09-18 18:49:39.463562315 +0200
@@ -26,6 +26,14 @@ def start(args, unlocked_cb=None):
             lines.append("Name=" + appInfo["name"])
             lines.append("Exec=waydroid app launch " + packageName)
             lines.append("Icon=" + args.waydroid_data + "/icons/" + packageName + ".png")
+            noShowList = ['com.android.calculator2', 'com.android.camera2', 'com.android.contacts', 
+                          'com.android.deskclock', 'com.android.documentsui', 'org.lineageos.etar',
+                          'com.android.email', 'com.android.gallery3d', 'com.android.inputmethod.latin', 
+                          'com.android.settings', 'org.lineageos.eleven', 'org.lineageos.jelly', 'org.lineageos.recorder']
+            if os.path.exists('/usr/bin/phosh') and packageName in noShowList:
+                lines.append("NoDisplay=true")
+            elif os.path.exists('/usr/bin/phosh'):
+                lines.append("X-Purism-FormFactor=Workstation;Mobile;")
             desktop_file = open(desktop_file_path, "w")
             for line in lines:
                 desktop_file.write(line + "\n")
@@ -44,6 +46,8 @@ def start(args, unlocked_cb=None):
         if hide:
             lines.append("NoDisplay=true")
         lines.append("Icon=" + tools.config.tools_src + "/data/AppIcon.png")
+        if os.path.exists('/usr/bin/phosh'):
+            lines.append("X-Purism-FormFactor=Workstation;Mobile;")
         desktop_file = open(desktop_file_path, "w")
         for line in lines:
             desktop_file.write(line + "\n")
